File: lab-2.3-ByLiza/C#1/MyString2.cs
using System;
using System.Globalization;

public class MyString2 : IKeyGenerator
{
    private string content;

    public string Content
    {
        get { return content + " 2"; }
    }

    public MyString2(string input)
    {
        content = input;
    }

    public string GenerateKey()
    {
        string[] words = content.Split(new string[] { " " }, StringSplitOptions.RemoveEmptyEntries);
        string key = "";

        for (int i = 0; i < words.Length; i++)
        {
            key += words[words.Length-i-1][0];
        }

        return key;
    }

    public void CapitalizeWords()
    {
        string[] words = content.Split(new string[] { " " }, StringSplitOptions.RemoveEmptyEntries);
        for (int i = 0; i < words.Length; i++)
        {
            if (words[i].Length > 0)
            {
                words[i] = char.ToLower(words[i][0]) + words[i].Substring(1);
            }
        }
        content = string.Join(" ", words);
    }
}

File: lab-2.3-ByLiza/C++/MyString2.cpp
#include "MyString2.h"
#include <sstream>
#include <cctype>
#include <vector>

MyString2::MyString2(const std::string& input) : content(input) {}

std::string MyString2::GetContent() const {
    return content + " 2";
}

std::string MyString2::GenerateKey() const {
    std::istringstream iss(content);
    std::vector<std::string> words;
    std::string word, key;
    while (iss >> word) {
        words.push_back(word);
    }
    for (int i = words.size() - 1; i >= 0; --i) {
        if (!words[i].empty())
            key += words[i][0];
    }
    return key;
}

void MyString2::CapitalizeWords() {
    std::istringstream iss(content);
    std::ostringstream oss;
    std::string word;
    bool first = true;
    while (iss >> word) {
        if (!word.empty())
            word[0] = std::tolower(word[0]);
        if (!first) oss << " ";
        oss << word;
        first = false;
    }
    content = oss.str();
}